node/src/hopper/routing_service.rs

```diff
@@ -1869,6 +1860,113 @@ mod tests {
         );
     }
 
+    fn route_data_to_peripheral_component_uses_proper_key_on_payload_for_component<F>(
+        payload_factory: F,
+        target_component: Component,
+    ) where
+        F: FnOnce(&CryptDEPair) -> CryptData,
+    {
+        let peer_actors = peer_actors_builder().build();
+        let subject = RoutingService::new(
+            make_cryptde_pair(),
+            RoutingServiceSubs {
+                proxy_client_subs_opt: peer_actors.proxy_client_opt,
+                proxy_server_subs: peer_actors.proxy_server,
+                neighborhood_subs: peer_actors.neighborhood,
+                hopper_subs: peer_actors.hopper,
+                to_dispatcher: peer_actors.dispatcher.from_dispatcher_client,
+                to_accountant_routing: peer_actors.accountant.report_routing_service_provided,
+            },
```

Comment:
You know... I'd like to see a utility function that would simply take `peer_actors` as the only argument and return complete RoutingServiceSubs.
You could use it here as well as at many other places in this file, speaking of tests. I do think it would be a good think to do. It takes too much interest for no big reason. 

`make_routing_service_subs_from_peer_actors()` or a better name...